### PR TITLE
[WIP] Add minikube dev env

### PR DESCRIPTION
**Pull request by petrkotas**
This PR introduces the alternative dev environment built on top of the minikube.
The main goal is to provide a simple dev environment as possible. 
This allows for easier developer onboarding.

@fabiand you were interested to discuss this approach

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/601)

### CODE REVIEW

**Comment by mykaul** [comment]
Performance will suck with
https://github.com/dhiltgen/docker-machine-kvm/pull/53

**Comment by petrkotas** [comment]
@mykaul Why exactly will the performance suck?

**Comment by mykaul** [comment]
On Thu, Nov 30, 2017 at 4:11 PM, Petr Kotas <notifications@github.com>
wrote:

> @mykaul <https://github.com/mykaul> Why exactly will the performance suck?
>

Minikube with current kvm driver runs qemu-kvm with the worst possible
devices.

File: cluster/minikube/mini_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2017 Red Hat, Inc.
+
+# get the config used widely in the kubevirt 
+docker_cmd_images="cmd/virt-controller cmd/virt-launcher cmd/virt-handler cmd/virt-api cmd/virt-manifest cmd/virt-dhcp cmd/virt-migrator"
```
Comment (davidvossel) [inline cluster/minikube/mini_deploy.sh]: converge with the configs in the hack folder. Make a config-minikube.sh if you need to.

File: cluster/minikube/mini_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    die "minikube is installed but not started. Please start minikube."
+  ok
+}
+
+_op_docker() {
+  parn "Building the docker images"
+
+  eval $(minikube docker-env)
+
+  # build docker images used by the kubevirt
```
Comment (davidvossel) [inline cluster/minikube/mini_deploy.sh]: we already have a docker build script, converge with that.

File: cluster/minikube/mini_deploy.sh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    (docker build --force-rm -t ${docker_prefix}/$(basename $arg):${docker_tag} -f $arg/Dockerfile.multi .)
+  done
+
+  ok
+}
+
+_build_manifests() {
+  parn "Building manifests"
+
+  pushd manifests/kubevirt
```
Comment (davidvossel) [inline cluster/minikube/mini_deploy.sh]: 'make manifests' converge with that. use env vars in hack/config-minikube.sh

File: cmd/virt-api/Dockerfile.mini [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#
```
Comment (davidvossel) [inline cmd/virt-api/Dockerfile.mini]: Why do you have new .mini Dockerfiles for every component?  This is more for us to maintain. Lets use the Dockerfile's we already have.

File: glide.lock [hunk 5]
```diff
@@ -1,87 +1,12 @@
 hash: b3ffe082966570570810f1f52c796da3dc31d24c84248da9aed9e06db26996ff
-updated: 2017-11-20T13:53:50.869715462-05:00
+updated: 2017-11-23T10:27:17.705741188+01:00
```
Comment (davidvossel) [inline glide.lock]: This patch series shouldn't require an update to glide.lock. No build dependencies have changed.

**Comment by simon3z** [comment]
@petrkotas @rmohr what are the requirements to introduce the multi-stage builds (`COPY --from=...`)?
IIUC they're supported on Docker 17.05 or higher but Fedora/CentOS ship Docker 13/12.
Does it mean that on Fedora/CentOS it would be required a third-party Docker-CE/Moby package?

**Comment by rmohr** [comment]
@simon3z I don't think we need multi-stage builds. You can also follow additional discussion on #609. For docker-only builds (where we don't want that one has to install any dependencies on the machine itself), a simple builder image, where we compile the sources is good enough.

**Comment by petrkotas** [comment]
@simon3z Roman is right. We actually do not need multistage builds. There is a cleaner way of building it.  There will be one docker image used for building stuff and the rest will just import the artifacts.

**Comment by rmohr** [comment]
That should be resolved by using the `PROVIDER=k8s-1.9.3`. It runs fully isolated within docker and also supports arbitrary amounts of nodes. @petrkotas maybe you want to try that provider out. Waiting for your feedback until I close this.

**Comment by rmohr** [comment]
Here more info: https://github.com/kubevirt/kubevirt/tree/master/cluster/k8s-1.9.3

**Pull request closed without merging**